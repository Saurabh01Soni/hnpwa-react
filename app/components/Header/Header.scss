.Header {
  background-color: $color-main;
  box-sizing: border-box;
  height: $header-height;
  padding: 18px 70px;
  width: 100%;
}

.Header__navlink {
  color: $color-white;
  margin: 0 10px;
  text-decoration: none;
}

.Header__navlink--active {
  text-decoration: underline;
  text-underline-position: under;
}

.Header__logo {
  float: right;
  height: 100%;
}

@media (max-width: 450px) {
  .Header {
    padding: 18px 10px;
  }
}

@media (max-width: 320px) {
  .Header__logo {
    display: none;
  }
}
