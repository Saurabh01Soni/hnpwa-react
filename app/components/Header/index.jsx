import React from 'react';
import { NavLink } from 'react-router-dom';

import './Header.scss';

function Header() {
  return (
    <nav className="Header">
      <NavLink
        activeClassName="Header__navlink--active"
        className="Header__navlink"
        to="/news"
      >
        Top
      </NavLink>
      <NavLink
        activeClassName="Header__navlink--active"
        className="Header__navlink"
        to="/newest"
      >
        New
      </NavLink>
      <NavLink
        activeClassName="Header__navlink--active"
        className="Header__navlink"
        to="/show"
      >
        Show
      </NavLink>
      <NavLink
        activeClassName="Header__navlink--active"
        className="Header__navlink"
        to="/ask"
      >
        Ask
      </NavLink>
      <NavLink
        activeClassName="Header__navlink--active"
        className="Header__navlink"
        to="/jobs"
      >
        Jobs
      </NavLink>
      <img
        className="Header__logo"
        src="https://raw.githubusercontent.com/tastejs/hacker-news-pwas/b3f3d40b9e4bd385dbb973d238ce207aed1f60eb/media/logo.png"
        alt="HNPWA Logo"
      />
    </nav>
  );
}

export default Header;
