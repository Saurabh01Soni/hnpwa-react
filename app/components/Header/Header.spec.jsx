import React from 'react';
import { shallow } from 'enzyme';

import Header, { makeIsActive } from './';

describe('<Header />', () => {
  it('should match snapshot when render default', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should make a function that returns true when location.pathname include path', () => {
    const newsIsActive = makeIsActive('/news');
    expect(newsIsActive(null, { pathname: '/news/1' })).toBe(true);
    expect(newsIsActive(null, { pathname: '/newest/1' })).toBe(false);

    const newestIsActive = makeIsActive('/newest');
    expect(newestIsActive(null, { pathname: '/newest/1' })).toBe(true);
    expect(newestIsActive(null, { pathname: '/show/1' })).toBe(false);
  });
});
