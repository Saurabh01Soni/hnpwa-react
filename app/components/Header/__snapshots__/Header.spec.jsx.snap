// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match snapshot when render default 1`] = `
<nav
  className="Header"
>
  <NavLink
    activeClassName="Header__navlink--active"
    ariaCurrent="true"
    className="Header__navlink"
    to="/news/1"
  >
    Top
  </NavLink>
  <NavLink
    activeClassName="Header__navlink--active"
    ariaCurrent="true"
    className="Header__navlink"
    to="/newest/1"
  >
    New
  </NavLink>
  <NavLink
    activeClassName="Header__navlink--active"
    ariaCurrent="true"
    className="Header__navlink"
    to="/show/1"
  >
    Show
  </NavLink>
  <NavLink
    activeClassName="Header__navlink--active"
    ariaCurrent="true"
    className="Header__navlink"
    to="/ask/1"
  >
    Ask
  </NavLink>
  <NavLink
    activeClassName="Header__navlink--active"
    ariaCurrent="true"
    className="Header__navlink"
    to="/jobs/1"
  >
    Jobs
  </NavLink>
  <img
    alt="HNPWA Logo"
    className="Header__logo"
    src="https://raw.githubusercontent.com/tastejs/hacker-news-pwas/b3f3d40b9e4bd385dbb973d238ce207aed1f60eb/media/logo.png"
  />
</nav>
`;
